src/utils/componentLibrary.ts

```diff
@@ -124,7 +127,9 @@ export async function populateComponentRefs<
     ref: ComponentReference,
   ): Promise<ComponentReference> {
     if (ref.spec) {
-      return ref;
+      const text = componentSpecToYaml(ref.spec);
```

Comment:
Why are we overwriting component text? This is a lossy conversion that we should avoid as much as possible?
Text should not be overwritten if it exists.

I think I've written the correct order of comparisons and linked the original source code.